cohttp-eio/src/server.ml

```diff
@@ -0,0 +1,260 @@
+        in
+        let source = Eio.Flow.cstruct_source iovecs in
+        Eio.Flow.copy source client_conn.oc;
+        (write [@tailcall]) ()
+    | `Yield ->
+        Eio.Std.Fibre.yield ();
+        (write [@tailcall]) ()
+    | `Close -> ()
+  in
+  Eio.Std.Fibre.both serialize write
+
+let rec handle_request (t : t) (conn : Client_connection.t) : unit =
+  match Reader.parse conn.reader Parser.request with
+  | req -> (
+      let read_complete = ref false in
+      let req = Request.{ req; reader = conn.reader; read_complete = false } in
+      match t.request_handler req with
+      | Some { Response.res; body } -> (
+          let keep_alive = Request.is_keep_alive req in
+          let response_headers =
+            Http.Header.add_unless_exists
+              (Http.Response.headers res)
+              "connection"
+              (if keep_alive then "keep-alive" else "close")
+          in
+          let res = { res with headers = response_headers } in
+          write_response conn { Response.res; body };
+          match keep_alive with
+          | false -> Client_connection.close conn
+          | true ->
+              (* Drain unread bytes from client connection before
+                 reading another request. *)
+              if not !read_complete then
+                match
+                  Http.Header.get_transfer_encoding (Request.headers req)
+                with
+                | Http.Transfer.Fixed _ -> ignore @@ Request.read_fixed req
+                | Http.Transfer.Chunked ->
+                    ignore @@ Request.read_chunk req ignore
+                | _ -> ()
+              else ();
+              (handle_request [@tailcall]) t conn)
+      | None ->
+          Printf.eprintf "Request not handled%!";
+          write_response conn Response.internal_server_error)
+  | exception Parser.Eof -> Client_connection.close conn
+  | exception Reader.Parse_error msg ->
+      Printf.eprintf "Request parsing error: %s" msg;
+      write_response conn Response.bad_request
+  | exception exn ->
+      Printf.eprintf "Unhandled exception: %s" (Printexc.to_string exn);
+      write_response conn Response.internal_server_error
+
+let run_accept_loop (t : t) sw env =
+  let net = Eio.Stdenv.net env in
+  let sockaddr = `Tcp (Eio.Net.Ipaddr.V4.loopback, t.port) in
+  let ssock =
+    Eio.Net.listen ~reuse_addr:true ~reuse_port:true ~backlog:t.socket_backlog
+      ~sw net sockaddr
+  in
+  let on_accept_error exn =
+    Printf.fprintf stderr "Error while accepting connection: %s"
+      (Printexc.to_string exn)
+  in
+  while not (Atomic.get t.closed) do
+    Eio.Net.accept_sub ~sw ssock ~on_error:on_accept_error
+    @@ fun ~sw flow addr ->
+    handle_request t
+      {
+        Client_connection.flow;
+        addr;
+        switch = sw;
+        reader = Reader.create (flow :> Eio.Flow.read);
+        oc = (flow :> Eio.Flow.write);
+      }
+  done
+
+let create ?(socket_backlog = 10_000) ?(domains = cpu_core_count) ~port
```

Comment:
What do you think about making the default for `domains` to be `1`? To me seems like the safest option since the user's code might not necessarily be thread safe.